doc: Describe supported output devices

mschwan-phytec opened the pull request:
Describe the supported output devices and their naming scheme in the documentation.

landerweit-phytec commented on doc/usage.rst:
```diff
@@ -20,3 +24,19 @@ When executing partup, the following options can be specified:
+Supported Output Devices
+------------------------
+
+Note, that writing is currently only supported for MMC devices. The device must
```
Maybe mention eMMC and the boot partitions here.

mschwan-phytec replied:
Done.

Added a more extensive list of supported devices.

mschwan-phytec commented:
I've updated the copyright year in an additional commit.

I've also added another paragraph noting that raw NAND or NOR devices are currently not supported.
